SmartCharger.Business/Services/EventService.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SmartCharger.Business.DTOs;
+using SmartCharger.Business.Interfaces;
+using SmartCharger.Data;
+using SmartCharger.Data.Entities;
+
+namespace SmartCharger.Business.Services
+{
+    public class EventService : GenericService<Event>, IEventService
+    {
+        public EventService(SmartChargerContext context) : base(context)
+        {
+        }
+
+        public async Task<EventResponseDTO> GetUsersChargingHistory(int userId, int page = 1, int pageSize = 5, string search = null)
+        {
+            try
+            {
+                var user = await _context.Users.SingleOrDefaultAsync(c => c.Id == userId);
+                if (user == null)
+                {
+                    return new EventResponseDTO
+                    {
+                        Success = false,
+                        Message = "User not found.",
+                        Error = "User not found."
+                    };
+                }
+                IQueryable<Event> query = _context.Events.Where(e => e.UserId == userId && e.EndTime != null);
+
+
+                if (!string.IsNullOrEmpty(search))
```

Comment:
Everything looks good, I just think search by first and last name isn't necessary.